compiler/bindings/python/iree/compiler/tools/import_onnx/__main__.py

```diff
@@ -145,17 +167,52 @@ def parse_arguments(argv=None) -> argparse.Namespace:
         " to before importing to MLIR. This can sometime assist with shape inference.",
         type=int,
     )
+    # args for saving a file with externalized params
+    parser.add_argument(
+        "--externalize-params",
+        help="Import the mlir file with large weights replaced by external reference calls.",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+    )
+    parser.add_argument(
+        "--externalize-inputs-threshold",
+        help="Treats inputs at or after the provided index as external parameters of the model."
+        " Only has an effect if 'externalize-params' is true.",
+        type=int,
+    )
+    parser.add_argument(
+        "--large-model",
+        help="Setting this to true is recommended for large models."
+        " It will bypass loading external weights and running the onnx checker to determine the model size.",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+    )
```

Comment:
The biggest issue I have is that inferring the model size (with weights) is not simple to do with onnx. 

I wanted to add this flag specifically so we could avoid running the checker, which is used simply to see if the model is >2GB in size. My issues with this checker is that it fails due to unrelated issues (e.g., duplicated metatdata_props) and that it actually hard crashes for some particularly large models. 

The other reason for adding this flag is to allow running the importer without loading the external onnx weights until the moment we want to save a param file.

Reply:
> The biggest issue I have is that inferring the model size (with weights) is not simple to do with onnx.

Can you check the file size? Not sure what I'm missing :P

Reply:
If the weights are external, you need to check the size of those files too. However, there isn't like a metadata field saying where all of the external weights are stored, so you'd need to iterate through the initializers and op attributes of the model to find all external data references, add them to a set, then check the disk usage of each of those files.

Reply:
Oof. Aren't the files at least downloaded in a group?

Found some docs:
* https://onnx.ai/onnx/repo-docs/ExternalData.html
* https://onnx.ai/onnx/repo-docs/PythonAPIOverview.html
* https://onnxruntime.ai/docs/tutorials/web/large-models.html

I'm missing a mental model for how users typically work with these "large" models 🤔  (2GB is not "large" in 2024 :P)

Reply:
2GB is a frustrating limit imposed by protobuf. 

It's only really problematic when trying to save a model without externalizing weights. A bit unrelated to the importer, but stuff like applying onnxruntime basic optimizations will attempt to save an optimized model, but sometimes the optimized model is larger than 2GB but the original isn't. You can apparently get the inference session to externalize the optimized model's weights, but how is one supposed to know when this is necessary? 

In general, the checker is going to take up a lot of time when the model is large, and there isn't a point in running it if we already know the model is large. Also, if a model is very large, we should avoid loading model weights unless totally necessary. 

In any case, if you find out a good way to say "hey onnx, how big is this model", please let me know and we can just replace the checker try/except block with an actual disk usage check.

Reply:
I think I'd need to go through some of the workflows myself with and without large external weights to form a good opinion. Can you point me to an example set of commands to run / models to try?

Here's another angle to think of this from: what happens if a user always sets `--large-model`? Are they missing out on some error checking? Will some optimizations not run? Are some features not supported? Based on the help text
```
        help="Setting this to true is recommended for large models."
        " It will bypass loading external weights and running the onnx checker to determine the model size.",
```
as a user, I might just set the flag no matter what... in which case how much are we gaining by making it default to off and including it at all?